internal/store/atlas/backup.go

```diff
@@ -17,39 +17,45 @@ package atlas
 import (
 	"fmt"
 
-	"github.com/mongodb/mongodb-atlas-cli/internal/config"
 	atlasv2 "go.mongodb.org/atlas-sdk/v20230201004/admin"
 )
 
-//go:generate mockgen -destination=../../mocks/atlas/mock_backup.go -package=atlas github.com/mongodb/mongodb-atlas-cli/internal/store/atlas CompliancePolicyDescriber,CompliancePolicy
-
+//go:generate mockgen -destination=../../mocks/atlas/mock_backup.go -package=atlas github.com/mongodb/mongodb-atlas-cli/internal/store/atlas CompliancePolicyDescriber,CompliancePolicy,CompliancePolicyEncryptionAtRestUpdater
 type CompliancePolicyDescriber interface {
 	DescribeCompliancePolicy(projectID string) (*atlasv2.DataProtectionSettings, error)
 }
+type CompliancePolicyUpdater interface {
+	UpdateCompliancePolicy(projectID string, opts *atlasv2.DataProtectionSettings) (*atlasv2.DataProtectionSettings, error)
+}
+type CompliancePolicyEncryptionAtRestUpdater interface {
+	UpdateEncryptionAtRest(projectID string, enable bool) (*atlasv2.DataProtectionSettings, error)
+}
 
 type CompliancePolicy interface {
-	DescribeCompliancePolicy(projectID string) (*atlasv2.DataProtectionSettings, error)
-	UpdateCompliancePolicy(projectID string, opts *atlasv2.DataProtectionSettings) (*atlasv2.DataProtectionSettings, error)
+	CompliancePolicyDescriber
+	CompliancePolicyUpdater
 }
 
-// DescribeCompliancePolicy encapsulates the logic to manage different cloud providers.
 func (s *Store) DescribeCompliancePolicy(projectID string) (*atlasv2.DataProtectionSettings, error) {
-	switch s.service {
-	case config.CloudService, config.CloudGovService:
-		result, _, err := s.clientv2.CloudBackupsApi.GetDataProtectionSettings(s.ctx, projectID).Execute()
-		return result, err
-	default:
-		return nil, fmt.Errorf("%w: %s", errUnsupportedService, s.service)
-	}
+	result, _, err := s.clientv2.CloudBackupsApi.GetDataProtectionSettings(s.ctx, projectID).Execute()
+	return result, err
 }
 
-// UpdateCompliancePolicy encapsulates the logic to manage different cloud providers.
 func (s *Store) UpdateCompliancePolicy(projectID string, opts *atlasv2.DataProtectionSettings) (*atlasv2.DataProtectionSettings, error) {
-	switch s.service {
-	case config.CloudService, config.CloudGovService:
-		result, _, err := s.clientv2.CloudBackupsApi.UpdateDataProtectionSettings(s.ctx, projectID, opts).Execute()
-		return result, err
-	default:
-		return nil, fmt.Errorf("%w: %s", errUnsupportedService, s.service)
+	result, _, err := s.clientv2.CloudBackupsApi.UpdateDataProtectionSettings(s.ctx, projectID, opts).Execute()
+	return result, err
+}
+
+func (s *Store) UpdateEncryptionAtRest(projectID string, enable bool) (*atlasv2.DataProtectionSettings, error) {
```

Comment:
strong preference to have a Disable and an Enable methods instead of a boolean arg

Reply:
Was unsure about this, is there a principle or something that speaks about this in more detail?

Reply:
I will need to try and find a good article but multiple points
In general passing a boolean to change behavior is considered a code smell and has to do with concepts of single responsibility and (your function does two things and depends on a param to pick one)